gtest/src/manager.rs

```diff
@@ -445,6 +449,61 @@ impl ExtManager {
 
         core_processor::handle_journal(journal, self);
     }
+
+    #[allow(dead_code)]
+    pub(crate) fn execute_custom_function(
+        &mut self,
+        source: ProgramId,
+        program_id: &ProgramId,
+        message_option: Option<IncomingMessage>,
+        function_name: &str,
+    ) -> Vec<u8> {
+        let mut executor = self.get_executor(source, program_id, message_option);
+        let (result, journal) = executor.call_function(function_name);
+        core_processor::handle_journal(journal, self);
+        result
+    }
+
+    #[allow(dead_code)]
+    pub(crate) fn execute_custom_void_function(
+        &mut self,
+        source: ProgramId,
+        program_id: &ProgramId,
+        message_option: Option<IncomingMessage>,
+        function_name: &str,
+    ) {
+        let mut executor = self.get_executor(source, program_id, message_option);
+        let journal = executor.call_void_function(function_name);
+        core_processor::handle_journal(journal, self);
+    }
+
+    fn get_executor(
+        &mut self,
+        source: ProgramId,
+        program_id: &ProgramId,
+        message_option: Option<IncomingMessage>,
+    ) -> WasmExecutor {
+        let (actor, balance) = self.actors.get_mut(program_id).unwrap();
+        if actor.is_uninitialized() {
+            actor.set_initialized();
+        }
+        if let Some(message) = message_option.clone() {
```

Comment:
Can we avoid clone here by borrowing `message_option`?

Reply:
Compiler don't let me to do that because we are borrowing it later to create wasm_executor instance

Reply:
You are confusing moving (changing ownership) and borrowing.

```rust
        if let Some(message) = &message_option {
            if message.source() != source {
                panic!("Source id in message: {:?} is not equal to source id in function arguments: {:?}", message.source(), source);
            }
        }
```